src/ariel-os-rt/build.rs

```diff
@@ -21,21 +21,22 @@ fn main() {
     // Put the linker scripts somewhere the linker can find them
     let out = &PathBuf::from(env::var_os("OUT_DIR").unwrap());
 
-    if let Some(context) = context_any(&["cortex-m", "riscv"]) {
-        let insert_before = match *context {
-            "riscv" => ".trap",
-            "cortex-m" => ".data",
-            _ => unreachable!(),
+    if let Some(context) = context_any(&["cortex-m", "riscv", "xtensa"]) {
+        let insert_somewhere = match *context {
+            "cortex-m" => "INSERT BEFORE .data;",
+            "riscv" => "INSERT BEFORE .trap;",
+            _ => "",
         };
 
         let region = match *context {
-            "riscv" => "RWDATA",
             "cortex-m" => "RAM",
+            "riscv" => "RWDATA",
+            "xtensa" => "RWDATA",
```

Comment:
```suggestion
            "riscv" | "xtensa" => "RWDATA",
```

Otherwise Clippy pedantic is going to complain when we unleash it at workspace-level (`pedantic` is enable on `ariel-os-rt` but this doesn't check build scripts).